### DISCUSSION TITLE
Jest tests are taking long time to run and failing unexpectedly after upgrading to version `1.7.17`.

### DISCUSSION THREAD

**Discussion by som-matrix** [tailwindlabs/headlessui#2832] [Help]
### Description

After upgrading to version `1.7.17` of `@headlessui/react`, Jest tests are experiencing unexpected behavior. The primary issue arises when attempting to mock `ResizeObserver` to resolve the "ReferenceError: ResizeObserver is not defined" error. However, adding the mock causes the tests to run very slowly and eventually fail.

### Versions

- `@headlessui/react`: `1.7.17`
- `Jest`: `^27.5.1`
- `Gatsby`: 4.23.0

### Steps to Reproduce

1. Upgrade `@headlessui/react` to version `1.7.17`.
2. Add a mock for `ResizeObserver` to resolve "ReferenceError: ResizeObserver is not defined." in setup-test-env.js file 
3. Run Jest tests.

### Expected Behavior

Jest tests should run successfully after upgrading to `@headlessui/react` version `1.7.17` with the mock for `ResizeObserver` without significant performance issues.

### Actual Behavior

Jest tests take an unusually long time to run and eventually fail after adding the mock for `ResizeObserver`.

### Environment
- Operating System: MacOs
- Node.js version: 16

Thanks for your assistance in resolving this issue!

**Comment by danny-avila** [comment]
Experiencing the same now, on @headlessui/react at 2.0.3

Will report back if I find a workaround

**Reply by danny-avila** [reply]
@som-matrix did you ever get around this?

**Reply by som-matrix** [reply]
Hey @danny-avila  , still can't believe this issue still persists. 
Workaround i did were as below:

1. In `setup-test-env.js` file i added this 
```
global.ResizeObserver = class ResizeObserver {
  observe() {}
  unobserve() {}
  disconnect() {}
};
```

2. In most of the tests , i have replaced `getByRole` with `getByText` and `getByRole` of type `textbox` with `getByLabelText` as it was taking long time to run for those tests and at the end it was failing.

**Comment by techwithanirudh** [comment]
Hi @RobinMalfait, sorry for the ping – could you take a look at this issue when you have a moment? Thanks!
